OpenRA.Mods.Common/UpdateRules/Rules/ConvertSupportPowerRangesToFootprint.cs

```diff
@@ -0,0 +1,95 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class ConvertSupportPowerRangesToFootprint : UpdateRule
+	{
+		public override string Name { get { return "Convert support power ranges to footprint"; } }
+		public override string Description
+		{
+			get
+			{
+				return "ChronoshiftPower and GrantExternalConditionPower use footprint areas\n" +
+				"instead of a circular range and they no longer have a fallback default area value.\n" +
+				"The old Range valuess will be converted to footprints as part of this update.";
+			}
+		}
+
+		static readonly string[] AffectedTraits = new string[] { "GrantExternalConditionPower", "ChronoshiftPower"};
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			foreach (var at in AffectedTraits)
+			{
+				foreach (var trait in actorNode.ChildrenMatching(at))
+				{
+					UpdatePower(trait);
+				}
+			}
+
+			yield break;
+		}
+
+		void UpdatePower(MiniYamlNode power)
+		{
+			var range = 1;
+			var rangeNode = power.LastChildMatching("Range");
+			if (rangeNode != null)
+			{
+				range = rangeNode.NodeValue<int>();
+				power.RemoveNode(rangeNode);
+			}
+
+			var size = 2 * range + 1;
+			power.AddNode(new MiniYamlNode("Dimensions", (size).ToString() + ", " + (size).ToString()));
+
+			string footprint = "";
+
+			int emptycell = range, affectedcell = 1;
+
+			for (int i = 0; i < size; i++)
+			{
+				for (int e = 0; e < emptycell; e++)
+				{
```

Comment:
Style nit (you can wait for more feedback/change requests before fixing this): we only use curly brackets around single-line statements if necessary for readability (which isn't the case here).